### DISCUSSION TITLE
Bandcamp support

### DISCUSSION THREAD

**Discussion by slackline** [MusicPlayerDaemon/MPD#2105] [Ideas]
[Bandcamp](https://bandcamp.com) is a popular platform for purchasing music that treats artists pretty well compared to other music services.

On purchasing music its possible to download copies in various formats but it also entitles users to unlimited streaming of the purchases.

I see there is plugin support for streaming from [qobuz](https://mpd.readthedocs.io/en/latest/plugins.html#qobuz) and was wondering whether its possible to adapt this approach to support streaming from Bandcamp?

**Comment by MaxKellermann** [comment]
Some sort of integration would certainly be possible, but I know nothing about Bandcamp's API so I can't say anything concrete. It would need a volunteer to work on this. Would you like to develop this?

**Comment by jcorporation** [comment]
It seems there is no public documented API available.

From https://bandcamp.com/developer:

```Labels and merchandise fulfillment partners can interact with their accounts via the Bandcamp API. ```

And from https://web.archive.org/web/20160512171900/http://bandcamp.com:80/developer:

```Sorry, the API is no longer supported and we're not granting any new developer keys.```

**Reply by MaxKellermann** [reply]
So they treat artists pretty well, but their customers not ;-)

**Reply by jcorporation** [reply]
It seems so…

**Reply by slackline** [reply]
Damn that sucks :disappointed: 

If I knew how to code in C++ I'd have a stab but alas I only know Python and R (a bit of Emacs Lisp and starting to dabble with Rust).

There are some python packages for interacting with Bandcamp (e.g. [bandcamper](https://github.com/yyyyyyyan/bandcamper))but a brief perusal doesn't reveal any that allow downloading of purchased music, most just look like they scrape what is freely available.
